File: benchmark/c/sum_loop.c
#include <stdint.h>
#include <stdio.h>

int main(void) {
    const int64_t iterations = 50000000;
    const int64_t mod_val = 2147483647;

    int64_t x = 1;
    int64_t acc = 0;

    for (int64_t i = 0; i < iterations; i++) {
        x = (x * 1664525 + 1013904223) % mod_val;
        acc += (x % 1024);
    }

    printf("%lld\n", (long long)acc);
    return 0;
}

File: benchmark/rust/sum_loop.rs
fn main() {
    let iterations: i64 = 50_000_000;
    let mod_val: i64 = 2_147_483_647;

    let mut x: i64 = 1;
    let mut acc: i64 = 0;

    for _ in 0..iterations {
        x = (x * 1_664_525 + 1_013_904_223) % mod_val;
        acc += x % 1024;
    }

    println!("{acc}");
}
